agent/hcp/manager.go

```diff
@@ -81,31 +84,38 @@ func NewManager(cfg ManagerConfig) *Manager {
 	}
 }
 
-// Run executes the Manager it's designed to be run in its own goroutine for
-// the life of a server agent. It should be run even if HCP is not configured
-// yet for servers since a config update might configure it later and
-// UpdateConfig called. It will effectively do nothing if there are no HCP
-// credentials set other than wait for some to be added.
-func (m *Manager) Run(ctx context.Context) error {
+// Start executes the logic for connecting to HCP and sending periodic server updates. If the
+// manager has been previously started, it will not start again.
+func (m *Manager) Start(ctx context.Context) error {
+	// Check if the manager has already started
+	if m.isRunning() {
+		m.logger.Trace("HCP manager already started")
+		return nil
+	}
+	m.setRunning(true)
```

Comment:
I think there's a possibility of a race here, even with the lock. Two goroutines can call `m.isRunning()` sequentially and both can see `false`, then both would proceed to call the rest of the code. I think we'd have to lock the lock, then do compare and set in one atomic operation like:

```golang
func (m *Manager) setRunningIfNotRunning() bool {
	m.runLock.Lock()
	defer m.runLock.Unlock()
        if m.running { return false }
	m.running = true
        return true
}
```

Reply:
Ah, great point, will make the change!